dataplane/Dockerfile

```diff
@@ -1,26 +1,40 @@
-FROM archlinux as builder
+FROM rust:1.75-slim-bookworm as builder
 
-RUN pacman -Syu --noconfirm
-RUN pacman -S base-devel protobuf rustup --noconfirm
+ARG TARGETARCH
+
+RUN apt-get update
+RUN apt-get install --yes build-essential protobuf-compiler pkg-config llvm-16
 
 RUN rustup default stable
 RUN rustup install nightly
 RUN rustup component add rust-src --toolchain nightly
-RUN rustup target add x86_64-unknown-linux-musl
 RUN --mount=type=cache,target=/root/.cargo/registry \
     cargo install bpf-linker
 
 WORKDIR /workspace
+# Docker uses the amd64/arm64 convention while Rust uses the x86_64/aarch64 convention.
+# Since Dockerfile doesn't support conditional variables (sigh), write the arch in Rust's
+# convention to a file for later usage.
+RUN if [ "$TARGETARCH" = "amd64" ]; \
+    then echo "x86_64" >> arch; \
+    else echo "aarch64" >> arch; \
+    fi
+RUN rustup target add $(eval cat arch)-unknown-linux-musl
```

Comment:
I really do wish everyone everwhere used the same naming convention for this but meh, what're you gonna do :shrug: